includes/admin/class-amp-editor-blocks.php

```diff
@@ -204,4 +212,110 @@ public function print_dirty_amp_scripts() {
 			wp_scripts()->do_items( $this->content_required_amp_scripts );
 		}
 	}
+
+	/**
+	 * Registers the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see register_block_core_latest_posts()
+	 */
+	public function register_block_latest_stories() {
+		register_block_type(
+			'amp/amp-latest-stories',
+			array(
+				'attributes'      => array(
+					'className'     => array(
+						'type' => 'string',
+					),
+					'storiesToShow' => array(
+						'type'    => 'number',
+						'default' => 5,
+					),
+					'order'         => array(
+						'type'    => 'string',
+						'default' => 'desc',
+					),
+					'orderBy'       => array(
+						'type'    => 'string',
+						'default' => 'date',
+					),
+				),
+				'render_callback' => array( $this, 'render_block_latest_stories' ),
+			)
+		);
+	}
+
+	/**
+	 * Renders the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see render_block_core_latest_posts()
+	 * @param array $attributes The block attributes.
+	 * @return string $markup The rendered block markup.
+	 */
+	public function render_block_latest_stories( $attributes ) {
+		$args = array(
+			'post_type'        => AMP_Story_Post_Type::POST_TYPE_SLUG,
+			'posts_per_page'   => $attributes['storiesToShow'],
+			'post_status'      => 'publish',
+			'order'            => $attributes['order'],
+			'orderby'          => $attributes['orderBy'],
+			'suppress_filters' => false,
+		);
+
+		$story_query = new WP_Query( $args );
+
+		ob_start();
+		foreach ( $story_query->posts as $post ) :
+			?>
+			<a href="<?php echo esc_url( get_permalink( $post ) ); ?>">
+				<?php echo get_the_post_thumbnail( $post->ID, self::LATEST_STORIES_IMAGE_SIZE ); ?>
+			</a>
+			<?php
+		endforeach;
+
+		$featured_images = ob_get_clean();
+		$class           = 'amp-block-latest-stories';
+
+		if ( isset( $attributes['className'] ) ) {
+			$class .= ' ' . $attributes['className'];
+		}
+
+		return sprintf(
+			'<amp-carousel height="%1$s" layout="fixed-height" type="slides" class="%2$s">%3$s</amp-carousel>',
+			esc_attr( $this->get_minimum_height( $story_query->posts ) ),
+			esc_attr( $class ),
+			$featured_images
+		);
+	}
+
+	/**
+	 * Gets the smallest height of the featured images of posts.
+	 *
+	 * Iterates through all of the posts, to find their featured image.
+	 * Then, this returns the smallest featured image height.
+	 *
+	 * @param array $posts An array or WP_Post objects.
+	 * @return int $minimum_height The smallest height of a featured image of any of the posts.
+	 */
+	public function get_minimum_height( $posts ) {
+		$minimum_height = 0;
+		$height_index   = 2;
+		foreach ( $posts as $post ) {
+			$image = wp_get_attachment_image_src( get_post_thumbnail_id( $post->ID ), self::LATEST_STORIES_IMAGE_SIZE );
```

Comment:
What if, by chance, a story doesn't have a featured image set?

Reply:
Good point. https://github.com/ampproject/amp-wp/pull/1916/commits/5e2c053f0eef3185900ed1c62dac0d69b295a165 continues in the loop if `get_post_thumbnail_id()` returns an empty string.